exchange/xgrid.F90

```diff
@@ -3272,6 +3262,7 @@ subroutine get_side1_from_xgrid(d, grid_id, x, xmap, complete)
   integer(i8_kind), dimension(MAX_FIELDS), save :: d_addrs=-9999
   integer(i8_kind), dimension(MAX_FIELDS), save :: x_addrs=-9999
 
+  d = 0
```

Comment:
2 Things here:

1. This seems very dangerous.  The variable comment for `d` says "recieved xgrid data".  Are you sure that the undefined behavior of the compilers was to return `d` as all 0?  I think a better idea would be to change `d` to `intent (inout)` or just `intent(in)` because it's never modified.  It's difficult to tell what the compiler actually does when this sort of thing is happening.
2. If `d` is a real, it should be
``` fortran
d = 0.
```

Reply:
I believe it's saying received because this function is receiving it. It's address gets sent to another routine where the data is set.

I think any compiler can return d with whatever happens to be in the address at the time (unless compiler flags are used to set all new values to 0) so this just sets it to 0 to make sure there's no random undefined values. I did this because it came up as an uninitialized warning but I made a similar change in #848, with gnu the undefined logical values were making the code fail so it needed to be initialized.